velox/exec/Task.cpp

```diff
@@ -168,6 +172,7 @@ Task::~Task() {
   } catch (const std::exception& e) {
     LOG(WARNING) << "Caught exception in ~Task(): " << e.what();
   }
+  ++numDeletedTasks_;
```

Comment:
Strictly speaking, this variable is updated before the task is destructed. Would that cause any problems?

Reply:
Yes, it can still happen in theory (especially if there is any blocking operations in the sub-class member destruction). But compared with the interference of the pending driver callback, it is very unlikely to happen.

Reply:
Would it make sense to create a small class that increments numCreatedTasks_ in the ctor and numDeletedTasks_ in the dtor, then declare a member variable of that class as the first member in the Task class? This way counters will be incremented first and last.